src/client/Client.cc

```diff
@@ -11236,8 +11232,16 @@ int64_t Client::_preadv_pwritev_locked(Fh *fh, const struct iovec *iov,
     if (fh->flags & O_PATH)
         return -CEPHFS_EBADF;
 #endif
+    if(iovcnt < 0) {
```

Comment:
Why not move this check to `Client::_preadv_pwritev`? Then you don't need to play with the locks.

Reply:
`Client::_preadv_pwritev` is not called in async io api (and it's only two callers i.e. `Client::preadv` and `Client::pwritev` already have this check), i think you meant `Client::ll_preadv_pwritev` and yes I agree this can be in it itself, and not only this but patches https://github.com/ceph/ceph/pull/54673 and https://github.com/ceph/ceph/pull/54672 too can just be handled in the caller

Reply:
i've modified the pr name to make it a bit more clear about the case im addressing i.e. the async io. the sync calls look fine.